x-pack/plugins/security_solution/server/lib/detection_engine/signals/executors/threshold.ts

```diff
@@ -71,136 +72,138 @@ export const thresholdExecutor = async ({
   let result = createSearchAfterReturnType();
   const ruleParams = completeRule.ruleParams;
 
-  // Get state or build initial state (on upgrade)
-  const { signalHistory, searchErrors: previousSearchErrors } = state.initialized
-    ? { signalHistory: state.signalHistory, searchErrors: [] }
-    : await getThresholdSignalHistory({
-        indexPattern: ['*'], // TODO: get outputIndex?
-        from: tuple.from.toISOString(),
-        to: tuple.to.toISOString(),
-        services,
-        logger,
-        ruleId: ruleParams.ruleId,
-        bucketByFields: ruleParams.threshold.field,
-        timestampOverride: ruleParams.timestampOverride,
-        buildRuleMessage,
-      });
-
-  if (!state.initialized) {
-    // Clean up any signal history that has fallen outside the window
-    const toDelete: string[] = [];
-    for (const [hash, entry] of Object.entries(signalHistory)) {
-      if (entry.lastSignalTimestamp < tuple.from.valueOf()) {
-        toDelete.push(hash);
+  return withSecuritySpan('detectionEngine thresholdExecutor', async () => {
```

Comment:
Just `thresholdExecutor`? Doesn't look like the other executors prefix with `detectionEngine`?

```suggestion
  return withSecuritySpan('thresholdExecutor', async () => {
```

Reply:
I previously used `detectionEngine` as a suffix but changed to `withSecuritySpan ` later to avoid duplication. Missed that piece during refactoring, thanks!